### PR TITLE
feat: Add default endpoint that forward POST requests to /webhooks

### PR DESCRIPTION
**Pull request by m-rudyk**
# Pull Request

## Description

In order to simplify setup for end user, default endpoint that forward requests added. Now user just need to paste LPVS URL in github webhooks without adding '/webhooks' endpoint. 
Corresponding guide updated.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [x] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [x] Test Coverage update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:
* Java: v17
* LPVS Release: v1.x.x

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My code meets the required code coverage for lines (90% and above)
- [x] My code meets the required code coverage for branches (80% and above)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by o-kopysov — COMMENTED** [review]
Please apply this patch to your PR:

``````
diff --git a/doc/docs/user-guide/service/database.md b/doc/docs/user-guide/service/database.md
index 85866d0..9963b91 100644
--- a/doc/docs/user-guide/service/database.md
+++ b/doc/docs/user-guide/service/database.md
@@ -32,7 +32,7 @@ sudo mysql

 !!! note

-    Replace 'username' and 'password' with your preferred values. However, in the example below, they are kept as placeholders for the sake of clarity:
+    Replace `username` and `password` with your preferred values. However, in the example below, they are kept as placeholders for the sake of clarity.

 ```sql
 create database lpvs;
diff --git a/doc/docs/user-guide/service/webhook.md b/doc/docs/user-guide/service/webhook.md
index 6a9048c..6f75f1c 100644
--- a/doc/docs/user-guide/service/webhook.md
+++ b/doc/docs/user-guide/service/webhook.md
@@ -44,7 +44,7 @@ Follow the next steps:

 ![step2](../../img/webhook/step_1_2.png)

-- Fill in the `Payload URL` with: <URL where LPVS is running>.
+- Fill in the `Payload URL` with URL where LPVS is running.

 !!! note

diff --git a/src/main/java/com/lpvs/controller/GitHubController.java b/src/main/java/com/lpvs/controller/GitHubController.java
index 649fcdc..99c4369 100644
--- a/src/main/java/com/lpvs/controller/GitHubController.java
+++ b/src/main/java/com/lpvs/controller/GitHubController.java
@@ -132,11 +132,10 @@ public class GitHubController {
     }

     /**
-     * @brief Default endpoint that forwards POST requests to the `/webhooks` endpoint.
-     *
+     * Default endpoint that forwards POST requests to the `/webhooks` endpoint.
      * This method serves as a "default" POST endpoint that doesn't have any specific path assigned.
      * It forwards any requests to the `/webhooks` endpoint using Spring's `RequestDispatcher`.
-     * Implemented to simplify Github webhook configuration.
+     * Implemented to simplify GitHub webhook configuration.
      *
      * @param request The HttpServletRequest object representing the incoming request.
      * @param response The HttpServletResponse object for the outgoing response.
@@ -146,7 +145,7 @@ public class GitHubController {
     @RequestMapping(method = RequestMethod.POST)
     public void forwardToWebhook(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
-        // Forward the request to the /webhooks endpoint
+        // Forward the request to the `/webhooks` endpoint
         RequestDispatcher dispatcher = request.getRequestDispatcher("/webhooks");
         dispatcher.forward(request, response);
     }
diff --git a/src/test/java/com/lpvs/controller/GitHubControllerTest.java b/src/test/java/com/lpvs/controller/GitHubControllerTest.java
index d2c4e3a..30b3192 100644
--- a/src/test/java/com/lpvs/controller/GitHubControllerTest.java
+++ b/src/test/java/com/lpvs/controller/GitHubControllerTest.java
@@ -81,17 +81,11 @@ public class GitHubControllerTest {

     @Test
     void ForwardToWebhookTest() throws ServletException, IOException {
-
         HttpServletRequest mockRequest = mock(HttpServletRequest.class);
         HttpServletResponse mockResponse = mock(HttpServletResponse.class);
         RequestDispatcher mockDispatcher = mock(RequestDispatcher.class);
-
         when(mockRequest.getRequestDispatcher("/webhooks")).thenReturn(mockDispatcher);
-
-        // Act
         gitHubController.forwardToWebhook(mockRequest, mockResponse);
-
-        // Assert
         verify(mockRequest).getRequestDispatcher("/webhooks");
         verify(mockDispatcher).forward(mockRequest, mockResponse);
     }

``````

**Comment by o-kopysov** [comment]
Also please update https://github.com/Samsung/LPVS/blob/main/doc/lpvs-api.yaml file

**Review by t-naumenko — APPROVED** [review]
LGTM

**Review by tdrozdovsky — DISMISSED** [review]

**Review by t-naumenko — APPROVED** [review]

File: doc/lpvs-api.yaml [hunk 1]
```diff
@@ -48,6 +48,26 @@ paths:
...
+            type: object
+            description: The request body is forwarded as-is to the `/webhooks` endpoint.
+    responses:
+      '200':
+        description: Request successfully forwarded and processed by `/webhooks`.
+      '403':
+        description: Forbidden, invalid signature or access rights.
+      '500':
+        description: Server error while forwarding the request.
+
```
Comment (o-kopysov) [inline doc/lpvs-api.yaml:70]: Please apply my patch:

``````
diff --git a/doc/lpvs-api.yaml b/doc/lpvs-api.yaml
index 34b67c8..c2ed547 100644
--- a/doc/lpvs-api.yaml
+++ b/doc/lpvs-api.yaml
@@ -48,6 +48,31 @@ paths:
               schema:
                 $ref: '#/components/schemas/WebhookResponseForbidden'

+  /:
+    post:
+      tags:
+        - GitHub Webhooks API
+      summary: Forward to Start Scan
+      description: Forwards the request to the `/webhooks` endpoint.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WebhookRequest'
+      responses:
+        '200':
+          description: 200 OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WebhookResponseOK'
+        '403':
+          description: 403 Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WebhookResponseForbidden'
+
   /scan/{gitHubOrg}/{gitHubRepo}/{prNumber}:
     post:
       tags:

``````

**Review by o-kopysov — APPROVED** [review]
LGTM

**Review by t-naumenko — APPROVED** [review]
LGTM

**Review by tdrozdovsky — APPROVED** [review]

**Pull request merged**